### PR TITLE
feat(plugin): FakeProfileThemes

### PR DESCRIPTION
**Pull request by remty5**
This plugin allows users to theme their profile by storing the colors in an invisible string in their bio.
This is a port of [Alyxia's Vendetta plugin](https://github.com/lexisother/vendetta-plugins/tree/master/plugins/hiddenProfileColours)

https://user-images.githubusercontent.com/42155711/227810001-bf04f0e5-1f9a-4356-9bcd-0feb987e1263.mp4

### CODE REVIEW

File: src/plugins/fakeProfileThemes.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    ],
+    settings,
+    colorDecodeHook(user: UserProfile) {
+        if (user) {
+            // don't replace colors if already set with nitro
+            if (settings.store.nitroFirst && user.themeColors) return user;
+            const colors = decode(user.bio);
+            if (colors) {
+                user.premiumType = 2;
+                user.themeColors = colors;
```
Comment (Vendicated) [inline src/plugins/fakeProfileThemes.tsx:103]: mutating the user in place like this is meh, maybe you could make this pure?
```js
return new Proxy(user, {
	get(target, prop) {
		switch (prop) {
			case "premiumType": return 2;
			case "themeColors": colors;
		}
		return target[prop];
	}
});
```

File: src/plugins/fakeProfileThemes.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    },
+    addCopy3y3Button(primary: number, accent: number) {
+        return <Button
+            onClick={() => {
+                const colorString = encode(primary, accent);
+                copyWithToast(colorString);
+            }}
+            color={Button.Colors.PRIMARY}
+            size={Button.Sizes.XLARGE}
+            style={{ marginLeft: "16px" }}
```
Comment (Vendicated) [inline src/plugins/fakeProfileThemes.tsx:116]:
```suggestion
            className={Margins.left16}
```

File: src/plugins/fakeProfileThemes.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            if (settings.store.nitroFirst && user.themeColors) return user;
+            const colors = decode(user.bio);
+            if (colors) {
+                user.premiumType = 2;
+                user.themeColors = colors;
+            }
+        }
+        return user;
+    },
+    addCopy3y3Button(primary: number, accent: number) {
```
Comment (Vendicated) [inline src/plugins/fakeProfileThemes.tsx:108]: Always wrap react stuffs in an ErrorBoundary to catch errors
```suggestion
    addCopy3y3Button: ErrorBoundary.wrap(function (primary: number, accent: number) {
```
(and close it at the end)

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Comment by Vendicated** [comment]
Also for future reference, you should never pr from the main branch! Make a feature branch for that

File: src/plugins/fakeProfileThemes.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+export default definePlugin({
+    name: "FakeProfileThemes",
+    description: "Allows profile theming by hiding the colors in your bio thanks to invisible 3y3 encoding.",
+    authors: [Devs.Alyxia, Devs.Remty],
+    patches: [
+        {
+            find: "getUserProfile=",
+            replacement: {
+                match: /(getUserProfile=function\(\i\){return )(\i\[\i\])/,
```
Comment (Vendicated) [inline src/plugins/fakeProfileThemes.tsx:92]:
```suggestion
                match: /(?<=getUserProfile=function\(\i\){return )(\i\[\i\])/,
```

No reason to match unchanged code

File: src/plugins/fakeProfileThemes.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                replace: "$1$self.colorDecodeHook($2)"
```
Comment (Vendicated) [inline src/plugins/fakeProfileThemes.tsx:93]:
```suggestion
                replace: "$self.colorDecodeHook($1)"
```

followup of above change

File: src/plugins/fakeProfileThemes.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+            }
+        }, {
+            find: ".USER_SETTINGS_PROFILE_THEME_ACCENT",
+            replacement: {
+                match: /ACCENT}\)}\)}\)(?<=(?<=},color:(\i).+)},color:(\i).+)/,
```
Comment (Vendicated) [inline src/plugins/fakeProfileThemes.tsx:98]: this is kind of messy, matching specific brackets is always only a last resort as it's not very robust, I will look more at this tomorrow

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Comment by Pixel556** [comment]
How to install?

**Comment by TheKodeToad** [comment]
If you wait it will probably be available in Vencord at some point unless there are issues.

**Review by Vendicated — APPROVED** [review]

**Comment by Vendicated** [comment]
Thank you!

**Comment by Im-IntoDev** [comment]
Can other people see it?

**Comment by Vendicated** [comment]
Anyone who has the plugin enabled will see it. Everyone else who doesn't will not see it

**Comment by Im-IntoDev** [comment]
Well then it's not so usefull, it's extremely cool and well thought out tho!

**Comment by Vendicated** [comment]
The other alternative would be to hack into Discord's servers and add the features to you but I fear that is not possible and also not very legal, so this is the best we can do

**Comment by somerandomcloud** [comment]
common bestdiscord L

**Comment by Im-IntoDev** [comment]
Well there's the better quality and emojis working without hacking into the servers.

**Comment by somerandomcloud** [comment]
Sounds like you arent actually into dev

**Comment by TheKodeToad** [comment]
xD it works by sending images

**Comment by Yudin101** [comment]
Umm....How do i enable it??
like i have the plugin enabled but it is not doing anything i can see a copy button beside it but idk

**Pull request merged**